tools/NewModuleScript/MainScreenSheetTemplate.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.displaycomposablemapview.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.BottomSheet
+import com.esri.arcgismaps.sample.sampleslib.components.DropDownMenuBox
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SamplePreviewSurface
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String) {
+    val mapViewModel: MapViewModel = viewModel()
+    var isBottomSheetVisible by remember { mutableStateOf(false) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        floatingActionButton = {
+            if (!isBottomSheetVisible) {
+                FloatingActionButton(
+                    modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { isBottomSheetVisible = true }
+                ) { Icon(Icons.Filled.Settings, contentDescription = "Show options") }
+            }
+        },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .weight(1f),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    onVisibleAreaChanged = { isBottomSheetVisible = false }
+                )
+            }
+
+            BottomSheet(
+                isVisible = isBottomSheetVisible,
+                sheetTitle = "Bottom sheet options",
+                onDismissRequest = { isBottomSheetVisible = false }
+            ) {
+                SampleOptions(
+                    // isCurrentOptionEnabled = ...,
+                    // onOptionToggled = { ... },
+                )
+            }
+
+            mapViewModel.messageDialogVM.apply {
+                if (dialogStatus) {
+                    MessageDialog(
+                        title = messageTitle,
+                        description = messageDescription,
+                        onDismissRequest = ::dismissDialog
+                    )
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun SampleOptions() {
+    Column(
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        DropDownMenuBox(
+            textFieldValue = "<selected-option>",
+            textFieldLabel = "Select an option",
+            dropDownItemList = emptyList(),
```

Comment:
As with the dialog options template - could have some dummy options

Reply:
Avoiding this as added above. :)